### PR TITLE
Improve loki boltdb shipper local and s3 storage setup

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR addresses two reliability issues when operating Loki with `boltdb-shipper` on S3 compatible storages beyond minio:
1. The ingester requires to be operated as a StatefulSet to ensure zero index file loss on replica restarts.
2. The S3 storage access is split to the required AWS environment variables instead of a simple URL endpoint.

For boltdb-shipper operations, consider [boltdb-shipper docs](https://grafana.com/docs/loki/latest/operations/storage/boltdb-shipper/#ingesters).

### CODE REVIEW

File: environments/base/default-config.libsonnet [hunk 1]
```diff
@@ -1,3 +1,4 @@
+local l = (import '../../components/loki.libsonnet');
```
Comment (squat) [inline environments/base/default-config.libsonnet:1]: Is this actually used?

File: operator/api/v1alpha1/observatorium_types.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ type ObjectStorageConfig struct {
 	Thanos *ObjectStorageConfigSpec `json:"thanos"`
```
Comment (squat) [inline operator/api/v1alpha1/observatorium_types.go:54]: Can we also have the corresponding name change here?

File: operator/api/v1alpha1/observatorium_types.go [hunk 3]
```diff
@@ -64,6 +64,26 @@ type ObjectStorageConfigSpec struct {
 	Key string `json:"key"`
 }
 
+type LokiObjectStorageConfigSpec struct {
+	// Object Store Config Secret Name
+	Name string `json:"name"`
```
Comment (squat) [inline operator/api/v1alpha1/observatorium_types.go:69]: Could we use `SecretName`? I think this helps communicate the expectation of the field, vs maybe suggesting we are naming this struct

File: environments/base/default-config.libsonnet (hunk 1 above)
Comment (periklis) [reply environments/base/default-config.libsonnet:1]: Nope!

File: operator/api/v1alpha1/observatorium_types.go (hunk 3 above)
Comment (periklis) [reply operator/api/v1alpha1/observatorium_types.go:69]: Will do, but I will leave the thanos config as is.

**Comment by periklis** [comment]
@squat Could you please take a look again.

**Review by squat — APPROVED** [review]
lgtm :))

**Pull request merged**